src/customer/strategies/googlepay/googlepay-authorizenet-customer-strategy.spec.ts

```diff
@@ -0,0 +1,258 @@
+        container.setAttribute('id', 'googlePayCheckoutButton');
+        walletButton = document.createElement('a');
+        walletButton.setAttribute('id', 'mockButton');
+
+        jest.spyOn(paymentProcessor, 'createButton')
+            .mockReturnValue(walletButton);
+
+        container.appendChild(walletButton);
+        document.body.appendChild(container);
+    });
+
+    afterEach(() => {
+        document.body.removeChild(container);
+    });
+
+    it('creates an instance of GooglePayCustomerStrategy', () => {
+        expect(strategy).toBeInstanceOf(GooglePayCustomerStrategy);
+    });
+
+    describe('#initialize()', () => {
+
+        describe('Payment method exist', () => {
+
+            it('Creates the button', async () => {
+                customerInitializeOptions = getAuthNetCustomerInitializeOptions();
+
+                await strategy.initialize(customerInitializeOptions);
+
+                expect(paymentProcessor.createButton).toHaveBeenCalled();
+            });
+
+            it('fails to initialize the strategy if no GooglePayCustomerInitializeOptions is provided ', () => {
+                customerInitializeOptions = getAuthNetCustomerInitializeOptions(Mode.Incomplete);
+
+                function initialize() {
+                    strategy.initialize(customerInitializeOptions);
+                }
+
+                expect(initialize).toThrow(InvalidArgumentError);
+            });
+
+            it('fails to initialize the strategy if no methodid is supplied', () => {
+                customerInitializeOptions = getAuthNetCustomerInitializeOptions(Mode.UndefinedMethodId);
+
+                function initialize() {
+                    strategy.initialize(customerInitializeOptions);
+                }
+
+                expect(initialize).toThrow(InvalidArgumentError);
+            });
+
+            it('fails to initialize the strategy if no valid container id is supplied', () => {
+                customerInitializeOptions = getAuthNetCustomerInitializeOptions(Mode.InvalidContainer);
+
+                return expect(strategy.initialize(customerInitializeOptions)).rejects.toThrow(InvalidArgumentError);
+            });
+        });
+    });
+
+    describe('#deinitialize()', () => {
+        let customerInitializeOptions: CustomerInitializeOptions;
+
+        beforeEach(() => {
+            customerInitializeOptions = getAuthNetCustomerInitializeOptions();
+        });
+
+        it('succesfully deinitializes the strategy', async () => {
+            await strategy.initialize(customerInitializeOptions);
+
+            strategy.deinitialize();
+
+            if (customerInitializeOptions.googlepayauthorizenet) {
+                const button = document.getElementById(customerInitializeOptions.googlepayauthorizenet.container);
+
+                if (button) {
+                    expect(button.firstChild).toBe(null);
+                }
+            }
+
+            // Prevent "After Each" failure
```

Comment:
can you add this block to an `afterEach` block instead of repeating the same thing in both tests?

Reply:
Refactored, no need for those lines of code, sorry.